### ISSUE TITLE
[Feature] Last used weapon command

### ISSUE THREAD

**Issue by eclenaeu** [ZDoom/gzdoom#1928]
### GZDoom version

Raze 1.6.2

### Which game are you running with GZDoom?

None

### What Operating System are you using?

None

### Is your feature request related to a problem? Please describe.

I don't know how other people play, but whenever I start an FPS style game, I use a light weapon and change settings to leave a hotkey for a heavy or explosive weapon (a grenade, rocket launcher or similar), to disperse groups of enemies or a stronger enemy, switching to the light weapon when possible to save ammo, using a command that quickly returns to the light weapon, this hotkey is Last used weapon, however, Raze does not have such a command, which bothers me makes having to use the previous weapon command, which is not so fast, or looking at the keyboard to locate the weapon key I want, but for those who play FPS games, taking their eyes off the screen or taking time to return shots, it can be the difference between winning or losing the fight.

### Describe the solution you'd like

the Last used weapon command would memorize the weapon used (remembering that, despite the term "light weapon", I could be talking about any gun you still have), allowing quick switching in combat, without taking your eyes off the screen, the command, despite not being original to the games supported by GZDoom, but it is present in other source ports of this game genre, however, I consider that GZDoom has superior performance and I believe that it would make the source port more complete.

**Comment by RicardoLuis0** [comment]
> Raze does not have such a command

This is the issues for GZDoom, you might want to post in https://github.com/ZDoom/raze/issues

**Comment by mjr4077au** [comment]
He's added a report for Raze, I think he wants it here as well but it was just a bad copy/pasta jobbie. If there's mods for it then all the better though. We're not at that point just yet, but inching closer

**Comment by eclenaeu** [comment]
> you might want to post in https://github.com/ZDoom/raze/issues

No, I don't, as mjr4077au suggested, it was just a copy/paste error, but with a clear objective: I posted in both projects because both don't have the command and also, I deduced that, because they are sister projects, this could increase the chances and interest to implement the feature in the source ports.

> And as for GZDoom, such a feature doesn't need any engine changes, since it's easily doable via modding

I completely disagree with you in that sense, I don't see how downloading a mod and configuring it to work can be easier than having the command natively implemented in the source port. If there is any challenge/difficulty to accomplish, that is another story. Weasel, in a forum, explained to me some difficulties to implement this in Raze, however I believe it could be easier in GZDoom (I'm just speculating), anyway, I'll correct the text so as not to generate any mistake about the purpose of the request and, leave the suggestion open for those who don't find it so difficult or unnecessary to be interested in the idea; which like I said, I believe will improve gameplay for many.
